packages/core/test/insured-bridge/BridgePool.js

```diff
@@ -742,6 +752,144 @@ describe("BridgePool", () => {
+          ev.relay.relayState === relayAttemptData.relayState
+        );
+      });
+      const speedupRelayStatus = await bridgePool.methods.relays(defaultDepositHash).call();
+      assert.equal(speedupRelayStatus, generateRelayHash(relayAttemptData));
+      const instantRelayHash = generateInstantRelayHash(defaultDepositHash, relayAttemptData);
+      assert.equal(await bridgePool.methods.instantRelays(instantRelayHash).call(), relayer);
+
+      // Check that contract pulled relay amount from instant relayer.
+      assert.equal(
+        (await l1Token.methods.balanceOf(relayer).call()).toString(),
+        "0",
+        "Instant Relayer should transfer relay amount"
+      );
+      assert.equal(
+        (await l1Token.methods.balanceOf(defaultDepositData.l1Recipient).call()).toString(),
+        instantRelayAmountSubFee,
+        "Recipient should receive the full amount, minus slow & instant fees"
+      );
+
+      // Grab proposed request data from events and submit for dispute:
+      const proposalEvent = (await optimisticOracle.getPastEvents("ProposePrice", { fromBlock: 0 }))[0];
+      await l1Token.methods.approve(optimisticOracle.options.address, totalRelayBond).send({ from: disputer });
+      await optimisticOracle.methods
+        .disputePrice(
+          bridgePool.options.address,
+          defaultIdentifier,
+          proposalEvent.returnValues.timestamp,
+          proposalEvent.returnValues.ancillaryData,
+          proposalEvent.returnValues.request
+        )
+        .send({ from: disputer });
+
+      // Cache price request timestamp.
+      await advanceTime(1);
+      const requestTimestamp = (await bridgePool.methods.getCurrentTime().call()).toString();
+      relayAttemptData = {
+        ...relayAttemptData,
+        slowRelayer: rando,
+        priceRequestTime: requestTimestamp,
+        relayId: (await bridgePool.methods.numberOfRelays().call()).toString(),
+        relayState: InsuredBridgeRelayStateEnum.PENDING,
+      };
+      const expectedExpirationTimestamp = (Number(requestTimestamp) + defaultLiveness).toString();
+
+      // Set up a relayAndSpeedUp transaction that should fail due to an existing relay.
+      await l1Token.methods.mint(rando, totalRelayBond.add(toBN(instantRelayAmountSubFee))).send({ from: owner });
+      await l1Token.methods
+        .approve(bridgePool.options.address, totalRelayBond.add(toBN(instantRelayAmountSubFee)))
+        .send({ from: rando });
+      await didContractThrow(
+        bridgePool.methods.relayAndSpeedUp(defaultDepositData, relayAttemptData.realizedLpFeePct).send({ from: rando })
+      );
+
+      // Reset params to allow for a normal relay.
+      await l1Token.methods.transfer(instantRelayer, instantRelayAmountSubFee).send({ from: rando });
+      await l1Token.methods.approve(bridgePool.options.address, totalRelayBond).send({ from: rando });
+
+      await bridgePool.methods.relayDeposit(...generateRelayParams()).send({ from: rando });
```

Comment:
nit: I think its more explicit if you call `relayDeposit` here with `...generateRelayParams({}, relayAttemptData)` to make it clear that you can slow relay the original, disputed relay, but you cannot `relayAndSpeedUp` or `speedUpRelay` it.

Reply:
Sounds good!